### PR TITLE
Add snakemake integration runner and kraken test

### PR DESCRIPTION
**Pull request by yesimon**
To test the entire pipes flow.

### CODE REVIEW

**Comment by dpark01** [comment]
Hm, I don't think this is going to work for various reasons. I think the better approach would be to:

1. Write basic integration tests that run kraken and krona on tiny test-case input (100 pairs of reads) without using snakemake. See test.unit.assembly for a few examples of things that arguably ought to be moved to test.integration (but we'll leave that for another time).  I really do want an execution of the actual binaries to be tested in a Py2.7 environment, which means it can't use snakemake.

2. See test.unit.test_snakemake.TestSimpleDryRuns for an example of something that tests the snakemake rules themselves, but skips the tests if Python < 3.2.

**Comment by yesimon** [comment]
I think we can test at a high level both the pipes and the python bins run via the command line, such that we make sure that argparse also works. This is the same test that was the original kraken "unit" test.

**Comment by dpark01** [comment]
That would be great so long as an execution test exists for 2.7 that does not rely on snakemake

File: test/integration/test_kraken.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            os.symlink(realpath, name)
+
+        # Minimizer len corresponds to memory/disk usage of index.
+        self.assertEqual(0, self.kraken.build(db, options={
+            '--minimizer-len': 10,
+            '--build': None,
+        }).returncode)
+        return db
+
+    def test_pipes(self):
```
Comment (dpark01) [inline test/integration/test_kraken.py]: Please add the decorator:
@unittest.skipIf(sys.version_info < (3, 2), "python version is too old for snakemake")

**Comment by yesimon** [comment]
In this I've modified TestCaseWithTmp to add a class level tmpdir for the purpose of caching a kraken db across multiple integration tests such that it can be reused.

**Comment by dpark01** [comment]
Nice... that's a useful add to that class.

**Comment by dpark01** [comment]
wtf--how long has github had like buttons and emoji's?? @tomkinsc

**Comment by dpark01** [comment]
from future import print command?

**Comment by tomkinsc** [comment]
About a month? It was in response to a [letter](https://github.com/dear-github/dear-github) complaining about `+1`s cluttering comment strings. Sadly the available "reactions" on GitHub don't include the full unicode emoji set...

**Pull request merged**